libcontainer/container_linux.go

```diff
@@ -1749,8 +1750,9 @@ func (c *linuxContainer) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Na
 					Value: []byte(c.newgidmapPath),
 				})
 			}
-			// The following only applies if we are root.
-			if !c.config.Rootless {
+			// The following only applies if we are root, unless we are
+			// advised to use the mapping tool.
+			if !(c.config.Rootless && !c.forceMappingTool) {
```

Comment:
What do you think of `if !c.config.Rootless || c.forceMappingTool`? Fewer `!`s is usually less confusing. I haven't run this, you should test it rather than take my word for it of course